### PR TITLE
Make OctoupsID configuration writable so our Cloud Platform can use it configure hosted Octopus Server instances

### PR DESCRIPTION
**Pull request by flin-8**
Our Cloud Platform currently uses the CLI to configure Octopus ID on a hosted instance. It would be quicker and more reliable to do this via the REST API.

We need to make these properties writable to facilitate this. Octopus Server will then enforce the appropriate permissions

https://trello.com/c/EHAyPCu1/690-improve-the-performance-of-configureinstanceaccess-step-in-customizehostedinstancebackgroundprocess

https://octopusdeploy.slack.com/archives/C01HZFJRYSH/p1643684744224069?thread_ts=1643590290.269429&cid=C01HZFJRYSH

### CODE REVIEW

**Review by zentron — APPROVED** [review]
The change looks reasonable to me from a F&M perspective, but id be keen to hear from @slewis74 if there were any string reasons for making them readonly in the first place apart from a bit of extra protection beyond permissions

**Review by tothegills — APPROVED** [review]

File: source/Server.OctopusID/Configuration/OctopusIDConfigurationResource.cs [hunk 1]
```diff
@@ -14,10 +15,12 @@ class OctopusIDConfigurationResource : OpenIDConnectConfigurationWithClientSecre
         /// NOTE: the following properties are here to control the order they appear on the settings page
         /// </summary>
 
-        [ReadOnly(true)]
+        [Writeable]
```
Comment (flin-8) [inline source/Server.OctopusID/Configuration/OctopusIDConfigurationResource.cs:18]: Removing `Readonly` wasn't enough, I needed to mark it as `Writable` too - I would've thought the attribute would be inherited 🤷

Comment (slewis74) [reply source/Server.OctopusID/Configuration/OctopusIDConfigurationResource.cs:18]: That's resource mapper, it doesn't do things without explicit instruction.

**Review by slewis74 — APPROVED** [review]
Seems reasonable to me, curious to see how server is enforcing the permissions 👍

**Comment by flin-8** [comment]
> Seems reasonable to me, curious to see how server is enforcing the permissions 👍

@slewis74 you can review here: https://github.com/OctopusDeploy/OctopusDeploy/pull/11572/files

**Pull request merged**